data_types/src/partition_metadata.rs

```
@@ -163,57 +163,27 @@ impl ColumnSummary {
-                s.count += o.count;
-                if o.min < s.min {
-                    s.min = o.min;
-                }
-                if o.max > s.max {
-                    s.max = o.max;
-                }
+                s.update_from(o);
             }
             (Statistics::I64(s), Statistics::I64(o)) => {
-                s.count += o.count;
-                if o.min < s.min {
-                    s.min = o.min;
-                }
-                if o.max > s.max {
-                    s.max = o.max;
-                }
+                s.update_from(o);
             }
             (Statistics::Bool(s), Statistics::Bool(o)) => {
-                s.count += o.count;
-                if s.min {
-                    s.min = o.min
-                }
-                if !s.max {
-                    s.max = o.max
-                }
+                s.update_from(o);
             }
             (Statistics::String(s), Statistics::String(o)) => {
-                s.count += o.count;
-                if o.min < s.min {
-                    s.min = o.min.clone();
-                }
-                if o.max > s.max {
-                    s.max = o.max.clone();
-                }
+                s.update_from(o);
             }
             (Statistics::U64(s), Statistics::U64(o)) => {
-                s.count += o.count;
-                if o.min < s.min {
-                    s.min = o.min;
-                }
-                if o.max > s.max {
-                    s.max = o.max;
-                }
+                s.update_from(o);
             }
             // do catch alls for the specific types, that way if a new type gets added, the compiler
             // will complain.
-            (Statistics::F64(_), _) => (),
-            (Statistics::I64(_), _) => (),
-            (Statistics::U64(_), _) => (),
-            (Statistics::Bool(_), _) => (),
-            (Statistics::String(_), _) => (),
+            (Statistics::F64(_), _) => unreachable!(),
```

I suspect we can hit this case (aka it is not `unreachable!` when data of different schema has been loaded into a different chunk: for example a field was sent initially with a `f64` value and then with an `I64` value into a different chunk that hasn't seen that field value before.

However, I think this change is an improvement because now we will know when/if it happens rather than just silently dropping the statistics